### PR TITLE
add releasing procedure doc

### PR DESCRIPTION
**Pull request by codeboten**
This is mostly pull from the opentelemetry-proto repository. Linked issue #46

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+Once deployment is done, go to Github [release
+page](https://github.com/open-telemetry/opentelemetry-configuration/releases), press
+`Draft a new release` to write release notes about the new release.
+
+You can use `git log upstream/v$MAJOR.$((MINOR-1)).x..upstream/v$MAJOR.$MINOR.x --graph --first-parent`
+or the Github [compare tool](https://github.com/open-telemetry/opentelemetry-configuration/compare/)
+to view a summary of all commits since last release as a reference.
+
+In addition, you can refer to [CHANGELOG.md](CHANGELOG.md)
```
Comment (jack-berg) [inline RELEASING.md:64]: Let's create a changelog!

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+for a list of major changes since last release.
+
+## Update release versions in documentations and CHANGELOG files
+
+After releasing is done, you need to update [README.md](README.md) and [CHANGELOG.md](CHANGELOG.md).
```
Comment (jack-berg) [inline RELEASING.md:69]: I don't actually think there's anything to update in the README. This actually appears to be true for `opentelemetry-proto` as well.

**Review by jack-berg — APPROVED** [review]
Only minor comments. Looks good 👍

File: RELEASING.md (hunk 1 above)
Comment (codeboten) [reply RELEASING.md:64]: Done

**Pull request merged**